File: 55MyPhotos/app/src/main/java/com/example/a55myphotos/MainActivity.java
package com.example.a55myphotos;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.os.Bundle;
import com.example.a55myphotos.model.Photo;
import com.example.a55myphotos.views.PhotoListAdaptor;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private PhotoViewModel photoViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        photoViewModel = new ViewModelProvider(this).get(PhotoViewModel.class);
        recyclerView = findViewById(R.id.recycler_View);

        photoViewModel.retrievePhotos();
        photoViewModel.photoListLiveData.observe(this, new Observer<List<Photo>>() {
            @Override
            public void onChanged(List<Photo> photoList) {
                createRecyclerView(photoList);
            }
        });
    }

    public void createRecyclerView(List<Photo> photoList) {
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
        PhotoListAdaptor adapter = new PhotoListAdaptor(photoList);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);
    }
}

File: MyPhotoKotlinVersion/app/src/main/java/com/example/myphotoskotlinversion/MainActivity.kt
package com.example.myphotoskotlinversion

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.myphotoskotlinversion.model.Photo
import com.example.myphotoskotlinversion.network.GetDataService
import com.example.myphotoskotlinversion.network.RetrofitClient
import com.example.myphotoskotlinversion.views.PhotoListAdapter
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.recycler_View)

        val getDataService = RetrofitClient.instance?.create(GetDataService::class.java)
        val callPhotos = getDataService?.getPhotos()
        callPhotos?.enqueue(object: Callback<MutableList<Photo>>{
            override fun onFailure(call: Call<MutableList<Photo>>, t: Throwable) {
                t.stackTrace
            }

            override fun onResponse(
                call: Call<MutableList<Photo>>,
                response: Response<MutableList<Photo>>
            ) {
             val photoList = response.body()
                if (photoList != null) {
                    for(photo in photoList){
                        Log.d("MyPhotos", "My photo: ${photo.title}")
                    }
                    createRecyclerView(photoList)
                }
            }

        })
    }

    fun createRecyclerView(photoList: MutableList<Photo>){
        val layoutManager = LinearLayoutManager(this)
        val adapter = PhotoListAdapter(photoList)
        recyclerView.layoutManager = layoutManager
        recyclerView.adapter = adapter
    }
}